src/environments/suweight.jl

```diff
@@ -370,3 +370,49 @@ function Base.rot180(wts::SUWeight)
     wts_y = circshift(rot180(wts[2, :, :]), (1, 0))
     return SUWeight(wts_x, wts_y)
 end
+
+function _CTMRGEnv(wts::SUWeight, flips::Array{Bool, 3})
+    @assert size(wts) == size(flips)
+    _, Nr, Nc = size(wts)
+    edges = map(Iterators.product(1:4, 1:Nr, 1:Nc)) do (d, r, c)
+        wt_idx = if d == NORTH
+            CartesianIndex(2, _next(r, Nr), c)
+        elseif d == EAST
+            CartesianIndex(1, r, _prev(c, Nc))
+        elseif d == SOUTH
+            CartesianIndex(2, r, c)
+        else # WEST
+            CartesianIndex(1, r, c)
+        end
+        wt = deepcopy(wts[wt_idx])
+        if (flips[wt_idx] && d in (NORTH, EAST)) || (!flips[wt_idx] && d in (SOUTH, WEST))
+            wt = permute(wt, ((2,), (1,)))
+        end
+        twistdual!(wt, 1)
+        wt = insertrightunit(insertleftunit(wt, 1))
```

Comment:
```suggestion
        wt = insertleftunit(insertleftunit(wt, 1), 4)
```

This should avoid having to permute (I think)

Reply:
Well, no. The results are the same:
```julia
julia> wt = DiagonalTensorMap(rand(2), ℂ^2)
DiagonalTensorMap(ℂ^2 ← ℂ^2):
 0.41815625829320335   ⋅ 
  ⋅                   0.04394804567655619


julia> insertrightunit(insertleftunit(wt, 1))
TensorMap((ℂ^1 ⊗ ℂ^2) ← (ℂ^2 ⊗ ℂ^1)):
[:, :, 1, 1] =
 0.41815625829320335  0.0

[:, :, 2, 1] =
 0.0  0.04394804567655619


julia> insertleftunit(insertleftunit(wt, 1), 4)
TensorMap((ℂ^1 ⊗ ℂ^2) ← (ℂ^2 ⊗ ℂ^1)):
[:, :, 1, 1] =
 0.41815625829320335  0.0

[:, :, 2, 1] =
 0.0  0.04394804567655619
```